core/play-integration-test/src/it/scala/play/it/http/websocket/WebSocketClient.scala

```
@@ -208,6 +210,11 @@ object WebSocketClient {
 
             def onPull(): Unit = pull(in)
 
+            override def onUpstreamFailure(ex: Throwable): Unit = {
+              logger.error("Communication problem using WebSocketClient", ex);
+              super.onUpstreamFailure(ex)
+            }
+
```

Logging, otherwise websocket client errors get swallowed. E.g. when trying to pass an invalid close status code to the client.